rust/kernel/error.rs

```diff
@@ -87,12 +99,15 @@ impl Error {
     pub(crate) unsafe fn from_kernel_errno_unchecked(errno: c_types::c_int) -> Error {
         // INVARIANT: the contract ensures the type invariant
         // will hold.
-        Error(errno)
+        //
+        // Safety: `errno` must not be zero, which is guaranteed by the contract
+        // of this function.
+        Error(unsafe { NonZeroI16::new_unchecked(errno as i16) })
```

Comment:
Does this need a `// CAST` annotation that explains _why_ casting a `c_int` to an `i16` won't lose any bits and result in UB?